Cython/Utility/ObjectHandling.c

```diff
@@ -2609,7 +2608,7 @@ static PyObject *__Pyx_PyVectorcall_FastCallDict_kw(PyObject *func, __pyx_vector
     }
 
     // The actual call
-    res = vc(func, newargs, nargs, kwnames);
+    res = vc(func, newargs, (size_t)nargs, kwnames);
```

Comment:
I'm not sure if all this is the right way round.

It looks to me like all the equivalent Python code uses `size_t`. I think that the only reason Cython uses `Py_ssize_t` is that it was an early adopter of this feature, and the initial drafts of the feature used that.

Therefore my opinion is that it would be better to change the function argument type to `size_t` and avoid casts in the body up-to-date version (but possibly have them in the code paths for the earlier "compatibility" versions).

I could be missing something here though, so it'd be good to have a second opinion before following my bad advice.

Reply:
Yes I thought about this as well, but was not sure, whether there was some thought behind the usage of Py_ssize_t in the interface. So I simply did it in the way that required less changes to get my build working ;)